### PR TITLE
Updated ERC20GatewayBase, ConsensusGateway and ConsensusCogateway.

### LINKED ISSUE #268: Update ERC20GatewayBase, ConsensusGateway and ConsensusCogateway.
- ERC20GatewayBase:
    - Update `DEPOSIT_INTENT_TYPEHASH` and `hashDepositIntent()` to include `address valueToken`.
    - Update `WITHDRAW_INTENT_TYPEHASH` and `hashWithdrawIntent()` to include `address valueToken,address utilityToken`.
- ConsensusGateway:
    - Update `deposit` function to pass most token in `hashDepositIntent()`.
    - Update `confirmWithdraw` function to pass most and utbase token address in `hashWithdrawIntent()`.
- ConsensusCogateway:
    - Update `withdraw` function to pass most and utbase token address in `hashWithdrawIntent()`.
    - Update `confirmDeposit` function to pass most address in `hashDepositIntent()`.

### PR DESCRIPTION
**Pull request by kreloaded**
Fixes #268 
- updated ERC20GatewayBase, ConsensusGateway and ConsensusCogateway.
- updated unit tests for ConsensusCogateway::confirmDeposit()
- updated unit tests for ERC20GatewayBase::hashWithdrawIntent()
- unit tests for ConsensusGateway::confirmWithdraw() are skipped now and will be handled in integration tests
     Reason : populating valueToken is necessary in proof file after generating proof. So that it can be used while testing ConsensusGateway::confirmWithdraw(). While testing valueToken is making a transaction hence, it can be handled in integration tests.

### CODE REVIEW

File: contracts/consensus-gateway/ConsensusGateway.sol [hunk 1]
```diff
@@ -41,6 +43,12 @@ contract ConsensusGateway is MasterCopyNonUpgradable, MessageBus, ConsensusGatew
     uint8 constant public INBOX_OFFSET = uint8(4);
 
 
+    /* Storage */
+
+    /* Utility token address. */
+    address public utilityToken;
```
Comment (deepesh-kn) [inline contracts/consensus-gateway/ConsensusGateway.sol]:
```suggestion
    address public utilityToken = address(0);
```

File: test/consensus-gateway/consensus_gateway/confirm_withdraw.js [hunk 2]
```diff
@@ -69,7 +69,7 @@ contract('ConsensusGateway::confirmWithdraw', (accounts) => {
   });
 
   contract('Positive Tests', () => {
-    it('should confirm withdraw', async () => {
+    it.skip('should confirm withdraw', async () => {
```
Comment (deepesh-kn) [inline test/consensus-gateway/consensus_gateway/confirm_withdraw.js:77]: Add comment why it is skipped so that everyone can know.

**Review by deepesh-kn — CHANGES_REQUESTED** [review]

File: contracts/test/consensus_gateway/ConsensusCogatewayDouble.sol [hunk 3]
```diff
@@ -54,5 +54,13 @@ contract ConsensusCogatewayDouble is ConsensusCogateway {
     function setMetablock(uint256 _metablockHeight) external {
         currentMetablockHeight = _metablockHeight;
     }
-}
 
+    /**
+     * It sets value token address.
```
Comment (gulshanvasnani) [inline contracts/test/consensus_gateway/ConsensusCogatewayDouble.sol]: Missing `@notice`.

File: contracts/consensus-gateway/ConsensusGateway.sol [hunk 4]
```diff
@@ -22,10 +22,12 @@ import "../core/CoreInterface.sol";
 import "../erc20-gateway/ERC20GatewayBase.sol";
 import "../message-bus/MessageBus.sol";
 import "../proxies/MasterCopyNonUpgradable.sol";
+import "../utility-token/UtilityTokenInterface.sol";
 
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 
-contract ConsensusGateway is MasterCopyNonUpgradable, MessageBus, ConsensusGatewayBase, ERC20GatewayBase, ConsensusModule, ConsensusGatewayInterface {
+contract ConsensusGateway is MasterCopyNonUpgradable, MessageBus, ConsensusGatewayBase,
+    ERC20GatewayBase,ConsensusModule, ConsensusGatewayInterface {
```
Comment (deepesh-kn) [inline contracts/consensus-gateway/ConsensusGateway.sol]:
```
contract ConsensusGateway is
    MasterCopyNonUpgradable,
    MessageBus,
    ConsensusGatewayBase,
    ERC20GatewayBase,
    ConsensusModule,
    ConsensusGatewayInterface {
```

**Review by deepesh-kn — CHANGES_REQUESTED** [review]

**Review by deepesh-kn — APPROVED** [review]

**Pull request merged**